phoenix-core/src/it/java/org/apache/phoenix/end2end/DropIndexedColsIT.java

```diff
@@ -0,0 +1,210 @@
+
+  @Parameterized.Parameters(name = "DropIndexedColsIT_salted={0}")
+  public static Collection<Boolean> data() {
+    return Arrays.asList(false, true);
+  }
+
+  @Test
+  public void testDropIndexedCols() throws Exception {
+    try (Connection conn = DriverManager.getConnection(getUrl());
+         Connection viewConn = DriverManager.getConnection(TENANT_SPECIFIC_URL)) {
+      String tableWithView1 = SchemaUtil.getTableName(SCHEMA1, generateUniqueName());
+      String tableWithView2 = SchemaUtil.getTableName(SCHEMA3, generateUniqueName());
+      String viewOfTable1 = SchemaUtil.getTableName(SCHEMA2, generateUniqueName());
+      String viewOfTable2 = SchemaUtil.getTableName(SCHEMA4, generateUniqueName());
+      String viewSchemaName1 = SchemaUtil.getSchemaNameFromFullName(viewOfTable1);
+      String viewSchemaName2 = SchemaUtil.getSchemaNameFromFullName(viewOfTable2);
+      String viewIndex1 = generateUniqueName();
+      String viewIndex2 = generateUniqueName();
+      String viewIndex3 = generateUniqueName();
+      String viewIndex4 = generateUniqueName();
+      String viewIndex5 = generateUniqueName();
+      String fullNameViewIndex1 = SchemaUtil.getTableName(viewSchemaName1, viewIndex1);
+      String fullNameViewIndex2 = SchemaUtil.getTableName(viewSchemaName1, viewIndex2);
+      String fullNameViewIndex3 = SchemaUtil.getTableName(viewSchemaName2, viewIndex3);
+      String fullNameViewIndex4 = SchemaUtil.getTableName(viewSchemaName2, viewIndex4);
+
+      conn.setAutoCommit(false);
+      viewConn.setAutoCommit(false);
+      String ddlFormat = new StringBuilder()
+          .append("CREATE TABLE ")
+          .append(tableWithView1)
+          .append(CREATE_TABLE_COL_QUERY).toString();
+      String ddlFormat2 = new StringBuilder()
+          .append("CREATE TABLE ")
+          .append(tableWithView2)
+          .append(CREATE_TABLE_COL_QUERY).toString();
+      conn.createStatement().execute(generateDDL(ddlFormat));
+      conn.createStatement().execute(generateDDL(ddlFormat2));
+      viewConn.createStatement().execute("CREATE VIEW " + viewOfTable1 +
+          " ( VIEW_COL1 DECIMAL(10,2), VIEW_COL2 VARCHAR ) AS SELECT * FROM " + tableWithView1);
+      viewConn.createStatement().execute("CREATE VIEW " + viewOfTable2 +
+          " ( VIEW_COL1 DECIMAL(10,2), VIEW_COL2 VARCHAR ) AS SELECT * FROM " + tableWithView2);
+
+      viewConn.createStatement().execute("CREATE INDEX " + viewIndex1 + " ON " + viewOfTable1
+          + "(v2) INCLUDE (v4)");
+      viewConn.createStatement().execute("CREATE INDEX " + viewIndex2 + " ON " + viewOfTable1
+          + "(v1) INCLUDE (v4)");
+      viewConn.createStatement().execute("CREATE INDEX " + viewIndex3 + " ON " + viewOfTable2
+          + "(v2) INCLUDE (v4)");
+      viewConn.createStatement().execute("CREATE INDEX " + viewIndex4 + " ON " + viewOfTable2
+          + "(v1) INCLUDE (v4)");
+      viewConn.createStatement().execute("CREATE INDEX " + viewIndex5 + " ON " + viewOfTable2
```

Comment:
nit: Can you extract these "CREATE TABLE/VIEW/INDEX" strings out as private static final member variables?